### PR TITLE
Routes changes 

### LINKED ISSUE #653: Student routes
```
/student/
    - List of courses
    - Redirect to /student/<course>/ if only one course

/student/<course>/
    - List of assignments

/student/<course>/assignments/ -> /student/<course>/

/student/<course>/assignments/<assignment>/
    - Assignment view
    - Shows group, first few backups, first few submissions, final submission
    - Can flag submissions

/student/<course>/assignments/<assignment>/backups/
/student/<course>/assignments/<assignment>/submissions/
    - Paginated list (?page=4) of backups and submissions
    - Can flag submissions here too

/student/<course>/assignments/<assignment>/backups/<id>/
/student/<course>/assignments/<assignment>/submissions/<id>/
    - View code for submission
```

### PR DESCRIPTION
**Pull request by Sumukh**
Resolves #653 

Should be okay to review.

### CODE REVIEW

File: server/controllers/main.py [hunk 1]
```diff
@@ -1,8 +1,13 @@
 from flask import Blueprint, render_template
 from server.extensions import cache
+from flask.ext.rq import get_queue
 
 main = Blueprint('main', __name__)
 
+def foo(i):
+    print(i)
```
Comment (Sumukh) [inline server/controllers/main.py]: To Remove (was testing RQ)

File: server/models.py [hunk 2]
```diff
@@ -245,17 +258,14 @@ class Message(db.Model, TimestampMixin):
...
     @hybrid_property
     def contents(self):
-        return json.dumps(str(self.raw_contents))
+        return json.loads(str(self.raw_contents))
 
     @contents.setter
     def contents(self, value):
         self.raw_contents = str(json.dumps(value))
 
-    def as_dict(self):
```
Comment (Sumukh) [inline server/models.py]: Should use `DictMixin` (but doesn't right now)

File: server/models.py [hunk 3]
```diff
@@ -163,6 +169,13 @@ def final_submission(self, user_ids):
             Backup.submit == True
         ).order_by(Backup.flagged.desc(), Backup.created.desc()).first()
 
+    def offering_name(self):
+        """ Returns the assignment name without the course offering.
+        """
+        return self.name.replace(self.course.offering + '/', '')
+
+
```
Comment (Sumukh) [inline server/models.py]: Too many new lines

File: server/templates/student/assignment/_tablehelper.html [hunk 4]
```diff
@@ -1,4 +1,4 @@
-{% macro render_subms(subms, cid, flagged, tname="Submissions", empty=True) %}
+{% macro render_subms(subms, course, flagged, tname="Submissions", empty=True) %}
```
Comment (Sumukh) [inline server/templates/student/assignment/_tablehelper.html]: We could just get away with only passing in the offering here.

**Comment by knrafto** [comment]
- [ ] None of the links work in `/submissions/` or `/backups/` (including the breadcrumb, "view code", and clicking the time)
- [ ] `/submissions/` and `/backups/` aren't accessible from anywhere. I suggest putting a link in the header of the backups and submissions tables in the assignment view, but I'm not sure how to style this
- [ ] Viewing a submission's code (`/code/<id>/`) shows "Backup 3" instead of "Submission 3"

**Comment by Sumukh** [comment]
image

**Comment by knrafto** [comment]
- [ ] Backups table after submissions table (not side-by-side)
- [ ] Only list non-submissions (submit=False) in `/backups/`
- [ ] Replace `/code/<id>/` with `/backups/<id>/` and `/submissions/<id>/`
- [ ] `/backups/<id>/` and `/submissions/<id>/` 404 for wrong backup type

**Pull request merged**